src/jit/morph.cpp

```diff
@@ -7722,7 +7722,7 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
     {       
         GenTreePtr value = gtArgEntryByArgNum(call, 2)->node;
 
-        if (value->OperGet() == GT_CNS_INT && value->AsIntConCommon()->IconValue() == 0)
+        if (value->IsZero())
```

Comment:
This will be true if value is a floating point const zero.  We need to check that it is an int const only here?

IMO, the right abstractions is

IsIntZero()
IsLngZero() - defined only for 32-bit targets
IsFloatZero() 
IsIntegralZero() - returns true if either of IsIntZero() or IsLngZero() is true
IsZero() - returns IsIntegralZero() || isFloatZero()

IsIntOne() - GT_CNS_INT and value is 1
IsLngOne() - defined only on 32-bit targets
IsIntegralOne() - returns IsIntOne() || IsLngOne()

IsIntMinusOne()
IsLngMinusOne()
IsIntegralMinusOne()

Now this check could be written as

`if (value->IsIntZero())`